ui/lib/components/ColumnsSelector/index.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { useState, ReactNode, useMemo } from 'react'
+import { Dropdown, Menu, Checkbox } from 'antd'
+import { DownOutlined } from '@ant-design/icons'
+import { useTranslation } from 'react-i18next'
+import { IColumn } from 'office-ui-fabric-react/lib/DetailsList'
+import { addTranslationResource } from '@lib/utils/i18n'
+
+const translations = {
+  en: {
+    text: 'Columns',
+  },
+  'zh-CN': {
+    text: '选择列',
+  },
+}
+
+for (const key in translations) {
+  addTranslationResource(key, {
+    component: {
+      columnsSelector: translations[key],
+    },
+  })
+}
+
+export interface IColumnsSelectorProps {
+  columns: IColumn[]
+  visibleColumnKeys?: { [key: string]: boolean }
+  onChange?: (visibleKeys: { [key: string]: boolean }) => void
+  headExtra?: ReactNode
+  footExtra?: ReactNode
+}
+
+export default function ColumnsSelector({
+  columns,
+  visibleColumnKeys,
+  onChange,
+  headExtra,
+  footExtra,
+}: IColumnsSelectorProps) {
+  const { t } = useTranslation()
+  const [dropdownVisible, setDropdownVisible] = useState(false)
+
+  const visibleKeys = useMemo(() => {
+    if (visibleColumnKeys) {
+      return visibleColumnKeys
+    }
+    return columns.reduce((acc, cur) => {
+      acc[cur.key] = true
+      return acc
+    }, {})
+  }, [visibleColumnKeys, columns])
+
+  const dropdownMenus = (
+    <Menu>
+      {headExtra && <Menu.Item key="head">{headExtra}</Menu.Item>}
+      {headExtra && <Menu.Divider />}
+
+      {columns
+        .filter((c) => c.key !== 'dummy')
+        .map((column) => (
+          <Menu.Item key={column.key}>
```

Comment:
BTW actually it might not be very good to use `<Menu.Item>` to render the column dropdown (or maybe the style of `Checkbox` need to be tuned), since there are many places that have hover actions but not clickable. 

![click](https://user-images.githubusercontent.com/1916485/80304742-992a3a80-87ea-11ea-8911-e4f5089419c6.gif)

Reply:
wow, so considerate, yep, if we can use `Select` to replace `Dropdown`. the Dropdown only can use `Menu` component as its overlay.

Reply:
Oh, actually I didn't mean to use `<Select>`. I think Dropdown is enough. My previous comment is saying that `<Select>` provides a custom render property. We can follow the same idea, providing a custom render property as well to render the tail item. In this way, the tail item is not limited to only one menu item, but can be anything.